instrumentation/kafka/src/main/java/brave/kafka/TracingConsumer.java

```diff
@@ -0,0 +1,181 @@
+package brave.kafka;
+
+import brave.Tracing;
+import org.apache.kafka.clients.consumer.*;
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+
+
+public class TracingConsumer<K, V> implements Consumer<K, V> {
```

Comment:
however this is exposed (via a decorating factory method or like this) needs type-level javadoc w/example